File: InvoiceReportForm.cs
using Serenity.ComponentModel;

namespace MyProject.Invoice.Forms
{
    public class InvoiceReportForm
    {
        [LookupEditor("YourVendorLookupKey")]
        public String Vendor { get; set; }

        [DateEditor]
        public DateTime StartDate { get; set; }

        [DateEditor]
        public DateTime EndDate { get; set; }
    }
}

File: InvoiceReportForm.ts
namespace MyProject.Invoice {
    export class InvoiceReportForm extends Serenity.PrefixedContext {
        static formKey = 'Invoice.InvoiceReport';

        constructor(prefix: string) {
            super(prefix);
        }

        public Vendor = Serenity.LookupEditor; 
        public StartDate = Serenity.DateEditor;
        public EndDate = Serenity.DateEditor;
    }
}
